.github/workflows/publish.yml

```diff
@@ -0,0 +1,67 @@
+on: [push, workflow_dispatch]
+
+jobs:
+  upload-images:
+    if: github.repository == 'iansjk/arknights-tools'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    name: Upload new images to Cloudinary
+    steps:
+      - name: Checkout recursively
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: Setup node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+
+      - name: Upload new images
+        run: |
+          yarn install
+          node scripts/images.mjs
+        env:
+          NEXT_PUBLIC_GOOGLE_ANALYTICS_ID: ${{ secrets.NEXT_PUBLIC_GOOGLE_ANALYTICS_ID }}
+          CLOUDINARY_CLOUD_NAME: ${{ secrets.CLOUDINARY_CLOUD_NAME }}
+          CLOUDINARY_API_KEY: ${{ secrets.CLOUDINARY_API_KEY }}
+          CLOUDINARY_API_SECRET: ${{ secrets.CLOUDINARY_API_SECRET }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      deployments: write
+    name: Deploy to Cloudflare Pages
+    steps:
+      - name: Checkout recursively
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: Setup node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+
+      - name: Build the website
+        run: |
+          yarn install
+          node scripts/update-data.mjs
+          yarn run build
+          yarn run next export
+        env:
+          NEXT_PUBLIC_GOOGLE_ANALYTICS_ID: ${{ secrets.NEXT_PUBLIC_GOOGLE_ANALYTICS_ID }}
+          CLOUDINARY_CLOUD_NAME: ${{ secrets.CLOUDINARY_CLOUD_NAME }}
+
+      - name: Publish to Cloudflare Pages
+        uses: cloudflare/pages-action@1
+        with:
+          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+          projectName: arknights-tools
+          directory: out
+          gitHubToken: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Bear with me since I'm not super well versed in Github Actions—is the intent that this workflow will run
- when manually triggered, or
- when anything is pushed to Github

And in response will
- run images.mjs
- run update-data.mjs
- publish the state of the repo after those changes to a Cloudflare Pages deployment?

(Regarding CFP preview vs production, the [documentation for `cloudflare/pages-action`](https://github.com/cloudflare/pages-action#specifying-a-branch) notes that
> The branch name is used by Cloudflare Pages to determine if the deployment is production or preview. 

So it should be a production deployment if we just pushed to `main` to trigger this workflow, and preview otherwise.)

But, this workflow feels a bit off to me—
- if e.g. I triggered this action manually on `main`, then it would update the images and gamedata, then immediately publish to the live site
  - ... but I often want to preview the update by creating a PR when I'm updating game data, to ensure that things work as expected with new characters (e.g. the recent addition of `Młynar` caused some unexpected issues for search that required additional work; similarly I tend to be more careful when a patch adds new materials, like the recent Ch11 update)
  - ... and opening a PR would already have the same effect of creating a preview deployment on CFP (because of the pre-existing CFP integration)
 
If I had to guess, I'd think that this workflow
- would be manually triggered, and
- would create a PR with the changes produced by running `images.mjs` and `update-data.mjs` (well, `images.mjs` doesn't modify the repo, but you know what I mean)

And as a result would create a preview deploy on CFP, since a new PR was created. Then if that preview looks right, the PR could be merged and a production deploy would be built and deployed on CFP automatically.

What do you think? Am I misunderstanding your intent somewhere?